driver/ppm_fillers.c

```diff
@@ -565,7 +576,11 @@ int accumulate_argv_or_env(const char __user * __user *argv,
 	for (;;) {
 		const char __user *p;
 
+#ifdef WDIG
+		if (ppm_copy_from_user(&p, argv, sizeof(p)) != 0)
+#else
 		if (unlikely(ppm_get_user(p, argv)))
+#endif
```

Comment:
That's the same thing, right? (if we don't remove the ppm_get_user definition for WDIG)

Reply:
No, because ppm_get_user is not available on Windows. The reason is that the macro, as defined, is not understood by Visual Studio. Instead of changing the macro, which felt dangerous, I changed the check here.

Reply:
Yeah, I suppose my question is why we can't have ppm_get_user on Windows. What's the problem and what's the macro definition that VS does like?

Reply:
it looks like it doesn't like the code block bracket in the macro
1>C:\GitHub\sysdig\driver\ppm_fillers.c(582,16): error C2059: syntax error: '{'
1>C:\GitHub\sysdig\driver\ppm_fillers.c(582,16): error C2143: syntax error: missing ';' before ')'
1>C:\GitHub\sysdig\driver\ppm_fillers.c(582,37): error C2059: syntax error: ')'
1>C:\GitHub\sysdig\driver\ppm_fillers.c(582,38): error C2059: syntax error: ')'

Reply:
Oh, right. `({ stmt; stmt })` is a GNU extension afaik, but it looks like it can be simply changed to `#define ppm_get_user(x, ptr) (ppm_copy_from_user(&x, ptr, sizeof(x)) ? -EFAULT : 0)`